File: src/main/java/Logic/Pet.java
package Logic;

public class Pet {
    private int fullness;
    private int cleanliness;
    private int energy;
    private int happiness;
    private String name;
    private int age;
    private String breed;

    public Pet(
            String name,
            int age,
            String breed,
            int currentFullness,
            int currentCleanliness,
            int currentEnergy,
            int currentHappiness) {
        if (currentFullness < 1 || currentFullness > 50) {
            throw new RuntimeException("Invalid fullness value " + currentFullness);
        }
        fullness = currentFullness;
        cleanliness = currentCleanliness;
        energy = currentEnergy;
        happiness = currentHappiness;
        this.name = name;
        this.age = age;
        this.breed = breed;
    }

    public int getFullness() {
        return fullness;
    }

    public void restoreFullness(int amount) {
        fullness += amount;
    }

    public void decreaseFullness(int amount) {
        fullness -= amount;
    }

    public int getCleanliness() {
        return cleanliness;
    }

    public void restoreCleanliness(int amount) {
        cleanliness += amount;
    }

    public void decreaseCleanliness(int amount) {
        cleanliness -= amount;
    }

    public int getEnergy() {
        return energy;
    }

    public void restoreEnergy(int amount) {
        energy += amount;
    }

    public void decreaseEnergy(int amount) {
        energy -= amount;
    }

    public int getHappiness() {
        return happiness;
    }

    public void restoreHappiness(int amount) {
        happiness += amount;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public String getBreed() {
        return breed;
    }
}

File: src/main/kotlin/Logic/Pet.kt
package Logic

class Pet(//primary constructor
    val name: String,
    var age: Int,
    var breed: String,
    currentFullness: Int,
    currentCleanliness: Int,
    currentEnergy: Int,
    currentHappiness: Int
) {//class' properties
    private var fullness: PetStat = PetStat(currentFullness)
    private var cleanliness: PetStat = PetStat(currentCleanliness)
    private var energy: PetStat = PetStat(currentEnergy)
    private var happiness: PetStat = PetStat(currentHappiness)

    fun getFullness(): Int {
        return fullness.getValue()
    }

    fun restoreFullness(amount: Int) {
        fullness.increaseValue(amount)
    }

    fun decreaseFullness(amount: Int) {
        fullness.decreaseValue(amount)
    }

    fun getCleanliness(): Int {
        return cleanliness.getValue()
    }

    fun restoreCleanliness(amount: Int) {
        cleanliness.increaseValue(amount)
    }

    fun decreaseCleanliness(amount: Int) {
        cleanliness.decreaseValue(amount)
    }

    fun getEnergy(): Int {
        return energy.getValue()
    }

    fun restoreEnergy(amount: Int) {
        energy.increaseValue(amount)
    }

    fun decreaseEnergy(amount: Int) {
        energy.decreaseValue(amount)
    }

    fun getHappiness(): Int {
        return happiness.getValue()
    }

    fun restoreHappiness(amount: Int) {
        happiness.increaseValue(amount)
    }

    fun decreaseHappiness(amount: Int) {
        happiness.decreaseValue(amount)
    }
}